models/offer.go

```diff
@@ -38,6 +42,37 @@ const enabledOffers = `
 		AND offers.enabled = true
 `
 
+func buildScope(enabledOffers string, filterAttrs map[string]string) string {
+	subQueries := []string{enabledOffers}
+	for k, v := range filterAttrs {
+		// TODO: Possible SQL injection
```

Comment:
Well, this part of the code is not protected, the caller is getAvailableOffers with the parameters from the query string. Avoiding single quotes is only a very simple safety, that I intend to improve before actually deploying this feature.